tests/vs-ldns.rs

```diff
@@ -0,0 +1,101 @@
+#![cfg(unix)]
+use std::process::Command;
+use tempfile::TempDir;
+
+#[ignore = "should only be run if ldns command line tools are installed"]
+#[test]
+fn nsec3_hash() {
+    assert_ldns_eq_dnst("ldns-nsec3-hash", &["example.com"]);
+    assert_ldns_eq_dnst("ldns-nsec3-hash", &["-a", "1", "example.com"]);
+    assert_ldns_eq_dnst("ldns-nsec3-hash", &["-s", "", "example.com"]);
+    assert_ldns_eq_dnst("ldns-nsec3-hash", &["-s", "DEADBEEF", "example.com"]);
+
+    for iterations in 0..10 {
+        assert_ldns_eq_dnst(
+            "ldns-nsec3-hash",
+            &["-t", &iterations.to_string(), "example.com"],
+        );
+    }
+
+    assert_exit_code_ldns_eq_dnst("ldns-nsec3-hash", &[]);
+}
+
+fn create_temp_symlink(ldns_command: &str) -> std::path::PathBuf {
```

Comment:
Ha this is clever. It's also a bit complicated though, maybe we should just assert that the ldns parsing creates the correct structs without a separate process?

Reply:
I don't like it at all and now that I think about it is shouldn't be needed.

We shouldn't test this here using real sym links, we should just test by passing the right argv0 to the outermost layer of our code, otherwise we're just testing O/S behaviour...

And for a sanity check that the behaviour when invoked as a sym link works we can do that in a Ploutos package test script which also validates that the right sym links are created on package install.

I'll remove this sym linking from the PR.

Reply:
And as you said a way to force test "as if" ldns would be nice and as you suggested we can add that as a separate bin that only differs in main by not including argv0 in the args passed to the argument parsing code.